packages/format-currency/src/index.ts

```diff
@@ -223,10 +261,17 @@ export function createFormatter(): CurrencyFormatter {
 		defaultLocale = locale;
 	}
 
+	// If the user is inside the US using USD, they should only see `$` and not `US$`.
+	async function detectGeolocation() {
```

Comment:
I think naming here is a bit confusing. 

When I see `@automattic/format-currency` exposing a `detectGeolocation()` function, it immediately feels weird - what does geo location detection has to do with currency formatting?

Once we read the function's contents, it all clicks. However, the function does something more on top of what it claims to do - it changes the currency symbol also.

So I'd suggest renaming this function to something more explicit, like `geolocateCurrencySymbol` for example. It could be a bit verbose, but it would remove the confusion I described above.

Reply:
Good idea! Naming is hard.